pool/cqrs_command/experiment_command.ml

```diff
@@ -145,16 +145,25 @@ end = struct
   (* Only when no sessions added *)
 
   type t =
-    { experiment_id : Id.t
+    { experiment : Experiment.t
     ; session_count : int
     }
 
-  let handle ?(tags = Logs.Tag.empty) { experiment_id; session_count } =
+  let handle ?(tags = Logs.Tag.empty) { experiment; session_count } =
     Logs.info ~src (fun m -> m "Handle command Delete" ~tags);
     match session_count > 0 with
     | true -> Error Pool_common.Message.ExperimentSessionCountNotZero
     | false ->
-      Ok [ Experiment.Destroyed experiment_id |> Pool_event.experiment ]
+      let filter_event =
+        experiment.Experiment.filter
+        |> CCOption.map_or ~default:[] (fun f ->
+             [ Filter.Deleted f |> Pool_event.filter ])
```

Comment:
isn't there a constructor for `Filter.deleted` --> `CCFun.(Filter.deleted %> Pool_event.filter %> CCList.pure`

Reply:
I added the variants functions